{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "7ff72ae0-d709-47f4-b7b7-fa8bf3c86ad7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/pedrofortesgonzalez/Desktop/MASTER_UNIR/31_TFM/3_repo/Anexo_3_tiab_screening\n"
     ]
    }
   ],
   "source": [
    "import os, time, pandas as pd               # importar librerías externas\n",
    "from Bio import Entrez\n",
    "from tqdm import tqdm\n",
    "import anexo_3_udfs as mf                   # importar funciones propias (UDFS)\n",
    "\n",
    "print(os.getcwd())                          # comprobar WD\n",
    "mf.viz()                                    # ampliar visualizaciones de pandas"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2004ed55-7719-4306-9665-d90f4f196556",
   "metadata": {},
   "source": [
    "# 1.cargar y de-duplicado"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "148f1608-8454-440c-962d-d37b55d3f9f5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(103, 90)\n",
      "\n",
      "Index(['Key', 'Item Type', 'Publication Year', 'Author', 'Title',\n",
      "       'Publication Title', 'ISBN', 'ISSN', 'DOI', 'Url', 'Abstract Note',\n",
      "       'Date', 'Date Added', 'Date Modified', 'Access Date', 'Pages',\n",
      "       'Num Pages', 'Issue', 'Volume', 'Number Of Volumes',\n",
      "       'Journal Abbreviation', 'Short Title', 'Series', 'Series Number',\n",
      "       'Series Text', 'Series Title', 'Publisher', 'Place', 'Language',\n",
      "       'Rights', 'Type', 'Archive', 'Archive Location', 'Library Catalog',\n",
      "       'Call Number', 'Extra', 'Notes', 'File Attachments', 'Link Attachments',\n",
      "       'Manual Tags', 'Automatic Tags', 'Editor', 'Series Editor',\n",
      "       'Translator', 'Contributor', 'Attorney Agent', 'Book Author',\n",
      "       'Cast Member', 'Commenter', 'Composer', 'Cosponsor', 'Counsel',\n",
      "       'Interviewer', 'Producer', 'Recipient', 'Reviewed Author',\n",
      "       'Scriptwriter', 'Words By', 'Guest', 'Number', 'Edition',\n",
      "       'Running Time', 'Scale', 'Medium', 'Artwork Size', 'Filing Date',\n",
      "       'Application Number', 'Assignee', 'Issuing Authority', 'Country',\n",
      "       'Meeting Name', 'Conference Name', 'Court', 'References', 'Reporter',\n",
      "       'Legal Status', 'Priority Numbers', 'Programming Language', 'Version',\n",
      "       'System', 'Code', 'Code Number', 'Section', 'Session', 'Committee',\n",
      "       'History', 'Legislative Body', 'combined_text', 'decision',\n",
      "       'has_abstract'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "# Cargar CSV de-duplicado\n",
    "path1 = \"./Anexo_3_carpeta_1_inputs/tabla_3.0_articulos_pre_revision.csv\"\n",
    "df = pd.read_csv(path1)\n",
    "print(df.shape)\n",
    "print(f\"\\n{df.columns}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1dd5c650-35be-4eba-ba9a-0c73bb0f97f9",
   "metadata": {},
   "source": [
    "Añadir numero de duplicados a stats"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3867dc65-e685-467e-9b42-21df8b9cc68a",
   "metadata": {},
   "source": [
    "### config para screening asistido"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "8f55b6c7-1515-4b4d-a727-6967f4b7d044",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "No. papers: 103\n",
      "Papers w/ abstract in Zotero: 97\n",
      "Papers w/out abstract in Zotero: 6\n"
     ]
    }
   ],
   "source": [
    "# comprobar cuántos artículos con abstract tenemos\n",
    "col = \"Abstract Note\"\n",
    "nonan_values = df[col].value_counts(dropna=True)\n",
    "nan_values = df[col].value_counts(dropna=False).iloc[0]\n",
    "\n",
    "print(f\"\\nNo. papers: {len(df)}\")\n",
    "print(f\"Papers w/ abstract in Zotero: {len(nonan_values)}\")\n",
    "print(f\"Papers w/out abstract in Zotero: {nan_values}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "966d8b94-00c8-490e-b75e-190953400388",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Artículos sin abstract: 6 de 103\n"
     ]
    }
   ],
   "source": [
    "# Preprocesamiento básico\n",
    "df['combined_text'] = df['Title'].fillna('') + ' ' + df['Abstract Note'].fillna('')\n",
    "df['combined_text'] = df['combined_text'].str.lower()\n",
    "\n",
    "# Inicializar columna de decisiones\n",
    "df['decision'] = None  # Posibles valores: 'include', 'exclude', None (pendiente)\n",
    "\n",
    "# Modificación del preprocesamiento para manejar abstracts faltantes\n",
    "df['Title'] = df['Title'].fillna('')\n",
    "df['Abstract Note'] = df['Abstract Note'].fillna('')\n",
    "df['has_abstract'] = ~df['Abstract Note'].str.strip().eq('')\n",
    "\n",
    "# Crear una bandera para identificar registros sin abstract\n",
    "print(f\"Artículos sin abstract: {sum(~df['has_abstract'])} de {len(df)}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "09cdd18c-722a-4b97-8223-cdb3f40683c2",
   "metadata": {},
   "source": [
    "# 2.recuperación automática de abstracts faltantes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9afe03b3-12b3-416b-b43d-1be991670887",
   "metadata": {},
   "source": [
    "## 2.1.recuperación automática de abstracts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "2d629c40-7ef1-4892-a459-28fdbfc0ffb2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Recuperando abstracts:   0%|                              | 0/6 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error en Unpaywall para DOI 10.1681/ASN.2024xrre28mc: HTTPSConnectionPool(host='api.unpaywall.org', port=443): Max retries exceeded with url: /v2/10.1681/ASN.2024xrre28mc?email=pedro.fortes.gonzalez@sergas.es (Caused by SSLError(SSLEOFError(8, '[SSL: UNEXPECTED_EOF_WHILE_READING] EOF occurred in violation of protocol (_ssl.c:1007)')))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Recuperando abstracts:  17%|███▋                  | 1/6 [00:04<00:20,  4.18s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error en Semantic Scholar para DOI 10.1681/ASN.2024xrre28mc: \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Recuperando abstracts:  33%|███████▎              | 2/6 [00:06<00:11,  2.86s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error en Semantic Scholar para DOI 10.1093/ndt/gfv167.3: Paper with id DOI:10.1093/ndt/gfv167.3 not found\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Recuperando abstracts:  50%|███████████           | 3/6 [00:08<00:08,  2.68s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error en Semantic Scholar para DOI 10.1681/ASN.20240vzj2pha: Internal Server Error\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Recuperando abstracts:  67%|██████████████▋       | 4/6 [00:10<00:04,  2.39s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error en Semantic Scholar para DOI 10.1681/ASN.2024jkdywd8g: Internal Server Error\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Recuperando abstracts: 100%|██████████████████████| 6/6 [00:14<00:00,  2.49s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error en Semantic Scholar para DOI 10.1016/S0016-5085(17)33807-6: Internal Server Error\n",
      "Abstracts recuperados: 0\n",
      "Artículos sin DOI: 0\n",
      "Artículos que siguen sin abstract: 6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "# Configuración para PubMed\n",
    "mail = \"pedro.fortes.gonzalez@sergas.es\"\n",
    "Entrez.email = mail\n",
    "\n",
    "# Recuperar abstracts faltantes usando DOI\n",
    "abstracts_recovered = 0\n",
    "articles_without_doi = 0\n",
    "\n",
    "# Crear una copia de los índices de artículos sin abstract\n",
    "indices_without_abstract = df[~df['has_abstract']].index.tolist()\n",
    "\n",
    "# Usar tqdm para mostrar progreso\n",
    "for idx in tqdm(indices_without_abstract, desc=\"Recuperando abstracts\"):\n",
    "    doi = df.loc[idx, 'DOI']\n",
    "    if pd.notnull(doi) and doi.strip() != '':\n",
    "        abstract = mf.fetch_abstract_by_doi(doi, mail)\n",
    "        if abstract and len(abstract.strip()) > 0:\n",
    "            #df.at[idx, 'abstract'] = abstract\n",
    "            df.at[idx, 'Abstract Note'] = abstract  # Actualizar también la columna original\n",
    "            df.at[idx, 'has_abstract'] = True\n",
    "            abstracts_recovered += 1\n",
    "            time.sleep(0.5)  # Respetar límites de API\n",
    "    else:\n",
    "        articles_without_doi += 1\n",
    "\n",
    "print(f\"Abstracts recuperados: {abstracts_recovered}\")\n",
    "print(f\"Artículos sin DOI: {articles_without_doi}\")\n",
    "print(f\"Artículos que siguen sin abstract: {sum(~df['has_abstract'])}\")\n",
    "\n",
    "# exportar resultado\n",
    "df.to_csv('Anexo_3_carpeta_1_inputs/tabla_3.0_articulos_pre_revision.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fb085a0d-614c-40af-88df-97530eac12d8",
   "metadata": {},
   "source": [
    "#### ¿cuántos abstracts hemos recuperado en automático?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "06baad08-24b9-49da-8bb2-f805a3b9a6d4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nº artículos CON abstract:\n",
      "has_abstract\n",
      "True     97\n",
      "False     6\n",
      "Name: count, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "col = \"has_abstract\"\n",
    "print(f\"Nº artículos CON abstract:\\n{df[col].value_counts(dropna=False)}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2fa770ef-b38c-4cac-80ac-30c39875b683",
   "metadata": {},
   "source": [
    "#### 1/7\n",
    "Voy a intentar recuperarlos manualmente. Para ello, primero separo el df en 2 (con y sin abstract) y los exporto para crear un checkpoint\n",
    "\n",
    "#### Separar df en df_with_abstract y df_without_abstract\n",
    "De acuerdo a columna \"has_abstract\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "8059e407-c44a-445a-9f53-84c92cdf602d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generado 'articulos_con_abstract.csv' para revisión manual\n",
      "Shape: (97, 90)\n",
      "Generado 'articulos_sin_abstract.csv' para revisión manual\n",
      "Shape: (6, 90)\n"
     ]
    }
   ],
   "source": [
    "# Separar artículos con y sin abstract\n",
    "df_with_abstract = df[df['has_abstract']].copy()\n",
    "df_without_abstract = df[~df['has_abstract']].copy()\n",
    "\n",
    "# Exportar lista de artículos CON abstract para revisión manual (si es necesaria) o recarga\n",
    "df_with_abstract.to_csv('Anexo_3_carpeta_1_inputs/tabla_3.1_articulos_con_abstract.csv', index=False)\n",
    "print(\"Generado 'articulos_con_abstract.csv' para revisión manual\")\n",
    "print(f\"Shape: {df_with_abstract.shape}\")\n",
    "\n",
    "# ídem pero para artículos SIN abstract\n",
    "df_without_abstract.to_csv('Anexo_3_carpeta_1_inputs/tabla_3.2.1_articulos_sin_abstract.csv', index=False)\n",
    "print(\"Generado 'articulos_sin_abstract.csv' para revisión manual\")\n",
    "print(f\"Shape: {df_without_abstract.shape}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "590ee94e-788b-462b-804b-a025c1e9bf47",
   "metadata": {},
   "source": [
    "## 2.2.Recuperación manual de abstracts\n",
    "\n",
    "Obtendré sus títulos y haré búsqueda avanzada en WOS y PubMed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "8e10e6e0-3cf7-4361-a271-cc65008b90ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "# subsetear cols (para facilitar proceso)\n",
    "cols = ['Author', 'Title', 'Publication Title', 'DOI', 'Abstract Note']\n",
    "df_with_abstract = df_with_abstract[df_with_abstract.columns.intersection(cols)]\n",
    "df_without_abstract = df_without_abstract[df_without_abstract.columns.intersection(cols)]\n",
    "\n",
    "# almaceno títulos de los papers en lista\n",
    "noabs_titles = df_without_abstract['Title'].value_counts(dropna=False).index.to_list()\n",
    "\n",
    "# inicio las queries como string vacía\n",
    "pm_query = \"\"\n",
    "wos_query= \"\"\n",
    "# defino variable de longitud total que quiero\n",
    "total = len(noabs_titles)-1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "90514a61-c811-472a-85c9-92aba65958cb",
   "metadata": {},
   "source": [
    "***\n",
    "#### query de PubMed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "b68ac5d5-06ba-43df-b032-e4ea3365f36e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total de títulos: 6\n",
      "\n",
      "PubMed Title query:\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "\"'Characterization of GANAB Phenotype in Patients with ADPKD Using Advanced Imaging Biomarkers'[tiab] OR 'URINARY PROTEOMIC BIOMARKERS, A POWERFUL TOOL FOR PROGNOSIS OF AUTOSOMAL DOMINANT POLYCYSTIC KIDNEY DISEASE PROGRESSION'[tiab] OR 'Characterization of IFT140 Phenotype in Patients with ADPKD Using Advanced Imaging Biomarkers'[tiab] OR 'Phenotypic Characterization of Patients with ADPKD with PKD1 and PKD2 Pathogenic Variants Using Advanced Imaging Biomarkers'[tiab] OR 'Addressing the gap in kidney care: A call for algorithmic approach with multiple-biomarker assessment'[tiab]\""
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# la construyo con un bucle\n",
    "for i,t in enumerate(noabs_titles):\n",
    "    if i == 0:                           # primer término\n",
    "        pm_query = f\"'{t}\"\n",
    "    elif i < total:                      # resto de términos (excp. último)\n",
    "        pm_query += (\"'[tiab] OR '\" + t)\n",
    "    elif i == total:                     # último término\n",
    "        pm_query += (\"'[tiab]\")\n",
    "\n",
    "# resultado\n",
    "print(f\"Total de títulos: {i+1}\")\n",
    "print(\"\\nPubMed Title query:\\n\")\n",
    "pm_query"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7799f78f-585d-49bd-aa99-aa7bcccde3a6",
   "metadata": {},
   "source": [
    "Search with this query retrieved 0 results\n",
    "\n",
    "***\n",
    "#### query de WOS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "61a33ff1-6c2a-4686-a783-9bc552dc80f3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total de títulos: 6\n",
      "\n",
      "WOS Title query:\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "\"TS=('Characterization of GANAB Phenotype in Patients with ADPKD Using Advanced Imaging Biomarkers' OR 'URINARY PROTEOMIC BIOMARKERS, A POWERFUL TOOL FOR PROGNOSIS OF AUTOSOMAL DOMINANT POLYCYSTIC KIDNEY DISEASE PROGRESSION' OR 'Characterization of IFT140 Phenotype in Patients with ADPKD Using Advanced Imaging Biomarkers' OR 'Phenotypic Characterization of Patients with ADPKD with PKD1 and PKD2 Pathogenic Variants Using Advanced Imaging Biomarkers' OR 'Addressing the gap in kidney care: A call for algorithmic approach with multiple-biomarker assessment')\""
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# la construyo con un bucle\n",
    "for i,t in enumerate(noabs_titles):\n",
    "    if i == 0:                           # primer término\n",
    "        pm_query = f\"TS=('{t}\"\n",
    "    elif i < total:                      # resto de términos (excp. último)\n",
    "        pm_query += (\"' OR '\" + t)\n",
    "    elif i == total:                     # último término\n",
    "        pm_query += (\"')\")\n",
    "\n",
    "# resultado\n",
    "print(f\"Total de títulos: {i+1}\")\n",
    "print(\"\\nWOS Title query:\\n\")\n",
    "pm_query"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "57bed0f8-117f-4d16-bd27-530704791c02",
   "metadata": {},
   "source": [
    "#### unir resultados con df\n",
    "\n",
    "importar recuperados, hacer concat entre ellos (abs_rec) y pd.merge con df (df, abs_rec, left_join)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "bdf40112-7c86-403b-b463-2aae8f01e03d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(97, 5)\n"
     ]
    }
   ],
   "source": [
    "# importar recuperados\n",
    "abs_rec_wos = pd.read_csv(\"Anexo_3_carpeta_1_inputs/tabla_3.2.2_abs_rec_wos.csv\", sep=\";\")\n",
    "#abs_rec_pm = pd.read_csv(\"tabla_3.2.3_abs_rec_pm.csv\", sep=\";\")  # no lo ejecutamos ya que no recuperamos ninguno\n",
    "\n",
    "# seleccionar subset de columnas\n",
    "pm_cols = [\"AUTHOR\", \"TITLE\", \"JOURNAL\", \"DOI\", \"ABSTRACT\"]\n",
    "wos_cols = ['AU', 'TI', 'SO', \"DL\", \"AB\"]\n",
    "abs_rec_wos = abs_rec_wos[abs_rec_wos.columns.intersection(wos_cols)]\n",
    "abs_rec_wos = abs_rec_wos[sorted(abs_rec_wos)]\n",
    "\n",
    "# renombrar cols\n",
    "abs_rec_wos.columns = [\"ABSTRACT\", \"AUTHOR\", \"DOI\", \"JOURNAL\", \"TITLE\"]\n",
    "# dropear duplicados\n",
    "abs_rec_wos = abs_rec_wos.drop_duplicates()\n",
    "\n",
    "# dropear nas en abstract\n",
    "abs_rec_wos = abs_rec_wos.dropna(subset=[\"ABSTRACT\"])\n",
    "\n",
    "## merge both into rec_abs\n",
    "rec_abs = abs_rec_wos.copy()\n",
    "# drop articles w/out abstract & duplicates\n",
    "rec_abs = rec_abs.drop_duplicates(subset=pm_cols[:-1])\n",
    "rec_abs = rec_abs.dropna(subset=[\"ABSTRACT\"])\n",
    "\n",
    "# capitalizar columnas del df y renombrar \"PUBLICATION TITLE\" > \"JOURNAL\" para facilitar pd.merge()\n",
    "df_with_abstract.columns = [col.upper() for col in list(df_with_abstract.columns)]\n",
    "df_with_abstract.columns = ['AUTHOR', 'TITLE', 'JOURNAL',  'DOI', 'ABSTRACT']\n",
    "\n",
    "## merge df & rec_abs\n",
    "rec_abs = pd.concat([rec_abs, df_with_abstract])\n",
    "rec_abs = rec_abs.drop_duplicates(subset=\"DOI\")  # drop articles w/out abstract & duplicates\n",
    "print(rec_abs.shape)                             # check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "bfcd331c-6ae8-4afe-883c-8442727662ea",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Artículos iniciales c. abstract: 97\n",
      "Artículos recuperados: 0\n",
      "Artículos finales c. abstract: 97\n"
     ]
    }
   ],
   "source": [
    "# check\n",
    "len_i = len(df_with_abstract)\n",
    "print(f\"Artículos iniciales c. abstract: {len_i}\")\n",
    "len_f = len(rec_abs)\n",
    "recuperados = len_i - len_f\n",
    "print(f\"Artículos recuperados: {recuperados}\")\n",
    "print(f\"Artículos finales c. abstract: {len_f}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0b19969e-0d73-418d-97de-7e108d74f556",
   "metadata": {},
   "source": [
    "Hemos recuperado 0 artículos buscando por TÍTULO, busquemos por DOI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "49a909e5-82cd-4f43-af76-1f16ef69575b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10.1681/ASN.2024xrre28mc\n",
      "10.1093/ndt/gfv167.3\n",
      "10.1681/ASN.20240vzj2pha\n",
      "10.1681/ASN.2024jkdywd8g\n",
      "10.4103/jfmpc.jfmpc_1120_24\n",
      "10.1016/S0016-5085(17)33807-6\n"
     ]
    }
   ],
   "source": [
    "# almaceno DOIS de los papers en lista\n",
    "noabs_dois = df_without_abstract['DOI'].value_counts(dropna=False).index.to_list()\n",
    "# print\n",
    "for doi in noabs_dois:\n",
    "    print(doi)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2e3cd510-3c81-4130-875a-b3f3a3480cc6",
   "metadata": {},
   "source": [
    "- En PM no obtengo nada (papers con otro DOI que dan como resultado este, y son Comentarios)\n",
    "- En WOS sí que aparecen, pero son 5 Meeting Abstracts y 1 Letter --> los añado a excluidos"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "7037f645-e5d3-47f8-9a17-4288863b3f34",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PubMed</th>\n",
       "      <th>WOS</th>\n",
       "      <th>Excluidos</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>#1: términos generales</th>\n",
       "      <td>150112</td>\n",
       "      <td>286496</td>\n",
       "      <td>0</td>\n",
       "      <td>436608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#2: patologías</th>\n",
       "      <td>506457</td>\n",
       "      <td>984086</td>\n",
       "      <td>0</td>\n",
       "      <td>1490543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#3: genes</th>\n",
       "      <td>2914</td>\n",
       "      <td>6025</td>\n",
       "      <td>0</td>\n",
       "      <td>8939</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#4: ómicas y biología computacional</th>\n",
       "      <td>2769404</td>\n",
       "      <td>11044214</td>\n",
       "      <td>0</td>\n",
       "      <td>13813618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#5: biomarcadores</th>\n",
       "      <td>1300101</td>\n",
       "      <td>1382132</td>\n",
       "      <td>0</td>\n",
       "      <td>2682233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Registros_identificados</th>\n",
       "      <td>211</td>\n",
       "      <td>49</td>\n",
       "      <td>0</td>\n",
       "      <td>260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Exclusion_cancer</th>\n",
       "      <td>172</td>\n",
       "      <td>45</td>\n",
       "      <td>43</td>\n",
       "      <td>217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Exclusion_last_10y</th>\n",
       "      <td>85</td>\n",
       "      <td>39</td>\n",
       "      <td>93</td>\n",
       "      <td>124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Exclusion_ingles</th>\n",
       "      <td>81</td>\n",
       "      <td>39</td>\n",
       "      <td>4</td>\n",
       "      <td>120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Duplicados y retractados</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Not papers</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      PubMed       WOS  Excluidos     Total\n",
       "#1: términos generales                150112    286496          0    436608\n",
       "#2: patologías                        506457    984086          0   1490543\n",
       "#3: genes                               2914      6025          0      8939\n",
       "#4: ómicas y biología computacional  2769404  11044214          0  13813618\n",
       "#5: biomarcadores                    1300101   1382132          0   2682233\n",
       "Registros_identificados                  211        49          0       260\n",
       "Exclusion_cancer                         172        45         43       217\n",
       "Exclusion_last_10y                        85        39         93       124\n",
       "Exclusion_ingles                          81        39          4       120\n",
       "Duplicados y retractados                   0         0         17       103\n",
       "Not papers                                 0         0          6        97"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# importar stats\n",
    "stats = pd.read_csv('../Anexo_2_busqueda_bbdd/Anexo_2_carpeta_2_resultados/tabla_2.4_search_stats.csv', index_col=0)\n",
    "\n",
    "# construir no_papers de deduplicaciones totales como dict (para luego concatenar)\n",
    "temp_df = {\"Not papers\": [0, 0, len(noabs_dois), len_f]}\n",
    "# pasar a df (y trasponer)\n",
    "temp_df = pd.DataFrame(temp_df).T.astype(int)\n",
    "# renombrar columnas\n",
    "temp_df.columns = [\"PubMed\", \"WOS\", \"Excluidos\", \"Total\"]\n",
    "\n",
    "# concatenar a stats\n",
    "stats = pd.concat([stats, temp_df], axis=0)\n",
    "stats"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e05832df-97ba-4e4c-9ab0-74f350c3a1c8",
   "metadata": {},
   "source": [
    "## 2.2.concatenar abstracts recuperados a articulos_con_abstract"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "499c435b-dfb1-402e-a923-d438350d4649",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape: (97, 90)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Key</th>\n",
       "      <th>Item Type</th>\n",
       "      <th>Publication Year</th>\n",
       "      <th>Author</th>\n",
       "      <th>Title</th>\n",
       "      <th>Publication Title</th>\n",
       "      <th>ISBN</th>\n",
       "      <th>ISSN</th>\n",
       "      <th>DOI</th>\n",
       "      <th>Url</th>\n",
       "      <th>Abstract Note</th>\n",
       "      <th>Date</th>\n",
       "      <th>Date Added</th>\n",
       "      <th>Date Modified</th>\n",
       "      <th>Access Date</th>\n",
       "      <th>Pages</th>\n",
       "      <th>Num Pages</th>\n",
       "      <th>Issue</th>\n",
       "      <th>Volume</th>\n",
       "      <th>Number Of Volumes</th>\n",
       "      <th>Journal Abbreviation</th>\n",
       "      <th>Short Title</th>\n",
       "      <th>Series</th>\n",
       "      <th>Series Number</th>\n",
       "      <th>Series Text</th>\n",
       "      <th>Series Title</th>\n",
       "      <th>Publisher</th>\n",
       "      <th>Place</th>\n",
       "      <th>Language</th>\n",
       "      <th>Rights</th>\n",
       "      <th>Type</th>\n",
       "      <th>Archive</th>\n",
       "      <th>Archive Location</th>\n",
       "      <th>Library Catalog</th>\n",
       "      <th>Call Number</th>\n",
       "      <th>Extra</th>\n",
       "      <th>Notes</th>\n",
       "      <th>File Attachments</th>\n",
       "      <th>Link Attachments</th>\n",
       "      <th>Manual Tags</th>\n",
       "      <th>Automatic Tags</th>\n",
       "      <th>Editor</th>\n",
       "      <th>Series Editor</th>\n",
       "      <th>Translator</th>\n",
       "      <th>Contributor</th>\n",
       "      <th>Attorney Agent</th>\n",
       "      <th>Book Author</th>\n",
       "      <th>Cast Member</th>\n",
       "      <th>Commenter</th>\n",
       "      <th>Composer</th>\n",
       "      <th>Cosponsor</th>\n",
       "      <th>Counsel</th>\n",
       "      <th>Interviewer</th>\n",
       "      <th>Producer</th>\n",
       "      <th>Recipient</th>\n",
       "      <th>Reviewed Author</th>\n",
       "      <th>Scriptwriter</th>\n",
       "      <th>Words By</th>\n",
       "      <th>Guest</th>\n",
       "      <th>Number</th>\n",
       "      <th>Edition</th>\n",
       "      <th>Running Time</th>\n",
       "      <th>Scale</th>\n",
       "      <th>Medium</th>\n",
       "      <th>Artwork Size</th>\n",
       "      <th>Filing Date</th>\n",
       "      <th>Application Number</th>\n",
       "      <th>Assignee</th>\n",
       "      <th>Issuing Authority</th>\n",
       "      <th>Country</th>\n",
       "      <th>Meeting Name</th>\n",
       "      <th>Conference Name</th>\n",
       "      <th>Court</th>\n",
       "      <th>References</th>\n",
       "      <th>Reporter</th>\n",
       "      <th>Legal Status</th>\n",
       "      <th>Priority Numbers</th>\n",
       "      <th>Programming Language</th>\n",
       "      <th>Version</th>\n",
       "      <th>System</th>\n",
       "      <th>Code</th>\n",
       "      <th>Code Number</th>\n",
       "      <th>Section</th>\n",
       "      <th>Session</th>\n",
       "      <th>Committee</th>\n",
       "      <th>History</th>\n",
       "      <th>Legislative Body</th>\n",
       "      <th>combined_text</th>\n",
       "      <th>decision</th>\n",
       "      <th>has_abstract</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>REREKIS5</td>\n",
       "      <td>journalArticle</td>\n",
       "      <td>2021</td>\n",
       "      <td>Stanzick, Kira J.; Li, Yong; Schlosser, Pascal; Gorski, Mathias; Wuttke, Matthias; Thomas, Laurent F.; Rasheed, Humaira; Rowan, Bryce X.; Graham, Sarah E.; Vanderweff, Brett R.; Patil, Snehal B.; Robinson-Cohen, Cassiane; Gaziano, John M.; O'Donnell, Christopher J.; Willer, Cristen J.; Hallan, Stein; Åsvold, Bjørn Olav; Gessner, Andre; Hung, Adriana M.; Pattaro, Cristian; Köttgen, Anna; Stark, Klaus J.; Heid, Iris M.; Winkler, Thomas W.</td>\n",
       "      <td>Discovery and prioritization of variants and genes for kidney function in &gt;1.2 million individuals.</td>\n",
       "      <td>Nature communications</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2041-1723</td>\n",
       "      <td>10.1038/s41467-021-24491-0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Genes underneath signals from genome-wide association studies (GWAS) for kidney function are promising targets for functional studies, but prioritizing variants  and genes is challenging. By GWAS meta-analysis for creatinine-based estimated  glomerular filtration rate (eGFR) from the Chronic Kidney Disease Genetics  Consortium and UK Biobank (n = 1,201,909), we expand the number of eGFRcrea loci  (424 loci, 201 novel; 9.8% eGFRcrea variance explained by 634 independent signal  variants). Our increased sample size in fine-mapping (n = 1,004,040, European)  more than doubles the number of signals with resolved fine-mapping (99% credible  sets down to 1 variant for 44 signals, ≤5 variants for 138 signals).  Cystatin-based eGFR and/or blood urea nitrogen association support 348 loci  (n = 460,826 and 852,678, respectively). Our customizable tool for Gene  PrioritiSation reveals 23 compelling genes including mechanistic insights and  enables navigation through genes and variants likely relevant for kidney function  in human to help select targets for experimental follow-up.</td>\n",
       "      <td>2021-07-16</td>\n",
       "      <td>2025-07-29 13:45:51</td>\n",
       "      <td>2025-07-29 13:45:51</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4350</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nat Commun</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>eng</td>\n",
       "      <td>© 2021. The Author(s).</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Place: England PMID: 34272381  PMCID: PMC8285412</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>*Genetic Predisposition to Disease; Biomarkers; Creatinine/blood; Cystatins/pharmacology; Databases, Genetic; Europe; Gene Expression Regulation/genetics; Genome-Wide Association Study; Glomerular Filtration Rate/*genetics; Humans; Kidney/*metabolism/physiology; Organ Specificity; Quantitative Trait Loci; Renal Insufficiency, Chronic/genetics/*metabolism; Risk Factors; RNA-Seq; Single-Cell Analysis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>discovery and prioritization of variants and genes for kidney function in &gt;1.2 million individuals. genes underneath signals from genome-wide association studies (gwas) for kidney function are promising targets for functional studies, but prioritizing variants  and genes is challenging. by gwas meta-analysis for creatinine-based estimated  glomerular filtration rate (egfr) from the chronic kidney disease genetics  consortium and uk biobank (n = 1,201,909), we expand the number of egfrcrea loci  (424 loci, 201 novel; 9.8% egfrcrea variance explained by 634 independent signal  variants). our increased sample size in fine-mapping (n = 1,004,040, european)  more than doubles the number of signals with resolved fine-mapping (99% credible  sets down to 1 variant for 44 signals, ≤5 variants for 138 signals).  cystatin-based egfr and/or blood urea nitrogen association support 348 loci  (n = 460,826 and 852,678, respectively). our customizable tool for gene  prioritisation reveals 23 compelling genes including mechanistic insights and  enables navigation through genes and variants likely relevant for kidney function  in human to help select targets for experimental follow-up.</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AS467S2X</td>\n",
       "      <td>journalArticle</td>\n",
       "      <td>2019</td>\n",
       "      <td>George, Jaya A.; Gounden, Verena</td>\n",
       "      <td>Novel glomerular filtration markers.</td>\n",
       "      <td>Advances in clinical chemistry</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0065-2423</td>\n",
       "      <td>10.1016/bs.acc.2018.10.005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chronic kidney disease is currently assessed by estimated glomerular filtration rate, a mathematical construct based on creatinine or creatinine and cystatin  concentration. Creatinine-based equations have improved with standardization  efforts and the Modification of Diet in Renal Disease Study (MDRD) and  CKD-Epidemiology Collaboration Study (CKD-EPI). Because the measurement of  creatinine is subject to interference from non-GFR determinants, alternative  markers have long been sought. These have included cystatin C and low molecular  weight proteins like β2-microglobulin and beta trace protein. Tubular disease  often occurs before glomerular filtration is impaired and investigators have  investigated the excretion of other low molecular weight proteins such as  Neutrophil Gelatinase-Associated Lipocalin (NGAL) and Kidney Injury Molecule-1  and N-acetyl-β-d-glucosaminidase. While preliminary, there is some evidence  linking these analytes with GFR, disease stage and mortality. Although  asymmetrical dimethyl arginine, an inhibitor of nitric oxide, has been shown to  be associated with progression of renal disease, symmetric dimethyl arginine may  be a better marker. Recent work has also explored the potential of microRNA  (miRNA) analysis and metabolomics studies to further elucidate this complex  pathophysiologic disease process. Investigators hope to improve our ability to  detect CKD by the use of test panels, i.e., various marker combinations thereof.  Unfortunately, most of these markers lack standardization unlike traditional  measures that rely on creatinine and cystatin C measurement.</td>\n",
       "      <td>2019</td>\n",
       "      <td>2025-07-29 13:45:51</td>\n",
       "      <td>2025-07-29 13:45:51</td>\n",
       "      <td>NaN</td>\n",
       "      <td>91-119</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Adv Clin Chem</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>eng</td>\n",
       "      <td>© 2019 Elsevier Inc. All rights reserved.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Place: United States PMID: 30612608</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Acetylglucosaminidase/analysis; Animals; Arginine/analogs &amp; derivatives/analysis; beta 2-Microglobulin/analysis; Biomarkers/analysis; Creatinine/analysis; Cystatin C; Cystatin C/analysis; GFR; Glomerular Filtration Rate; Hepatitis A Virus Cellular Receptor 1/analysis; Humans; Kidney/physiopathology; KIM-1; Lipocalin-2/analysis; miRNA; NGAL; Renal Insufficiency, Chronic/*diagnosis/physiopathology; β2-microglobulin</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>novel glomerular filtration markers. chronic kidney disease is currently assessed by estimated glomerular filtration rate, a mathematical construct based on creatinine or creatinine and cystatin  concentration. creatinine-based equations have improved with standardization  efforts and the modification of diet in renal disease study (mdrd) and  ckd-epidemiology collaboration study (ckd-epi). because the measurement of  creatinine is subject to interference from non-gfr determinants, alternative  markers have long been sought. these have included cystatin c and low molecular  weight proteins like β2-microglobulin and beta trace protein. tubular disease  often occurs before glomerular filtration is impaired and investigators have  investigated the excretion of other low molecular weight proteins such as  neutrophil gelatinase-associated lipocalin (ngal) and kidney injury molecule-1  and n-acetyl-β-d-glucosaminidase. while preliminary, there is some evidence  linking these analytes with gfr, disease stage and mortality. although  asymmetrical dimethyl arginine, an inhibitor of nitric oxide, has been shown to  be associated with progression of renal disease, symmetric dimethyl arginine may  be a better marker. recent work has also explored the potential of microrna  (mirna) analysis and metabolomics studies to further elucidate this complex  pathophysiologic disease process. investigators hope to improve our ability to  detect ckd by the use of test panels, i.e., various marker combinations thereof.  unfortunately, most of these markers lack standardization unlike traditional  measures that rely on creatinine and cystatin c measurement.</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>6LN8B6N4</td>\n",
       "      <td>journalArticle</td>\n",
       "      <td>2019</td>\n",
       "      <td>Capolongo, Giovanna; Zacchia, Miriam; Perna, Alessandra; Viggiano, Davide; Capasso, Giovambattista</td>\n",
       "      <td>Urinary proteome in inherited nephrolithiasis.</td>\n",
       "      <td>Urolithiasis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2194-7236 2194-7228</td>\n",
       "      <td>10.1007/s00240-018-01104-y</td>\n",
       "      <td>NaN</td>\n",
       "      <td>In the last decades, proteomics has been largely applied to the Nephrology field, with the double aim to (1) elucidate the biological processes underlying renal  diseases; (2) identify disease-specific biomarkers, predictor factors of  therapeutic efficacy and prognostic factors of disease progression. Kidney stone  disease, and in particular, inherited nephrolithiasis (INL) are not an exception.  Given the multifactorial origin of these disorders, the combination of genomics  and proteomics studies may complement each other, with the final objective to  give a global and comprehensive mechanistic view. In this review, we summarize  the results of recent proteomic studies which have expanded our knowledge about  INL, focusing the attention on monogenic forms of nephrolithiasis (cystinuria,  Dent's disease, Bartter syndrome, distal renal tubular acidosis and primary  hyperoxaluria), on polygenic hypercalciuria and on medullary sponge kidney  disease.</td>\n",
       "      <td>2019-02</td>\n",
       "      <td>2025-07-29 13:45:51</td>\n",
       "      <td>2025-07-29 13:45:51</td>\n",
       "      <td>NaN</td>\n",
       "      <td>91-98</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>47</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Urolithiasis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>eng</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Place: Germany PMID: 30564846</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Biomarker; Biomarkers/urine; Genetic nephrolithiasis; Humans; Hypercalciuria/diagnosis/genetics/*urine; Medullary Sponge Kidney/diagnosis/genetics/*urine; Nephrolithiasis/diagnosis/genetics/*urine; Proteome/*analysis; Proteomics; Proteomics/*methods; Urine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>urinary proteome in inherited nephrolithiasis. in the last decades, proteomics has been largely applied to the nephrology field, with the double aim to (1) elucidate the biological processes underlying renal  diseases; (2) identify disease-specific biomarkers, predictor factors of  therapeutic efficacy and prognostic factors of disease progression. kidney stone  disease, and in particular, inherited nephrolithiasis (inl) are not an exception.  given the multifactorial origin of these disorders, the combination of genomics  and proteomics studies may complement each other, with the final objective to  give a global and comprehensive mechanistic view. in this review, we summarize  the results of recent proteomic studies which have expanded our knowledge about  inl, focusing the attention on monogenic forms of nephrolithiasis (cystinuria,  dent's disease, bartter syndrome, distal renal tubular acidosis and primary  hyperoxaluria), on polygenic hypercalciuria and on medullary sponge kidney  disease.</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5S9M622R</td>\n",
       "      <td>journalArticle</td>\n",
       "      <td>2019</td>\n",
       "      <td>Woodard, Lauren E.; Welch, Richard C.; Veach, Ruth Ann; Beckermann, Thomas M.; Sha, Feng; Weinman, Edward J.; Ikizler, Talat Alp; Tischfield, Jay A.; Sahota, Amrik; Wilson, Matthew H.</td>\n",
       "      <td>Metabolic consequences of cystinuria.</td>\n",
       "      <td>BMC nephrology</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1471-2369</td>\n",
       "      <td>10.1186/s12882-019-1417-8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BACKGROUND: Cystinuria is an inherited disorder of renal amino acid transport that causes recurrent nephrolithiasis and significant morbidity in humans. It has  an incidence of 1 in 7000 worldwide making it one of the most common genetic  disorders in man. We phenotypically characterized a mouse model of cystinuria  type A resultant from knockout of Slc3a1. METHODS: Knockout of Slc3a1 at RNA and  protein levels was evaluated using real-time quantitative PCR and  immunofluorescence. Slc3a1 knockout mice were placed on normal or breeder chow  diets and evaluated for cystine stone formation over time suing x-ray analysis,  and the development of kidney injury by measuring injury biomarkers. Kidney  injury was also evaluated via histologic analysis. Amino acid levels were  measured in the blood of mice using high performance liquid chromatography. Liver  glutathione levels were measured using a luminescent-based assay. RESULTS: We  confirmed knockout of Slc3a1 at the RNA level, while Slc7a9 RNA representing the  co-transporter was preserved. As expected, we observed bladder stone formation in  Slc3a1(-/-) mice. Male Slc3a1(-/-) mice exhibited lower weights compared to  Slc3a1(+/+). Slc3a1(-/-) mice on a regular diet demonstrated elevated blood urea  nitrogen (BUN) without elevation of serum creatinine. However, placing the  knockout animals on a breeder chow diet, containing a higher cystine  concentration, resulted in the development of elevation of both BUN and  creatinine indicative of more severe chronic kidney disease. Histological  examination revealed that these dietary effects resulted in worsened kidney  tubular obstruction and interstitial inflammation as well as worsened bladder  inflammation. Cystine is a precursor for the antioxidant molecule glutathione, so  we evaluated glutathione levels in the livers of Slc3a1(-/-) mice. We found  significantly lowered levels of both reduced and total glutathione in the  knockout animals. CONCLUSIONS: Our results suggest that that diet can affect the  development and progression of chronic kidney disease in an animal model of  cystinuria, which may have important implications for patients with this disease.  Additionally, reduced glutathione may predispose those with cystinuria to injury  caused by oxidative stress. Word count: 327.</td>\n",
       "      <td>2019-06-20</td>\n",
       "      <td>2025-07-29 13:45:51</td>\n",
       "      <td>2025-07-29 13:45:51</td>\n",
       "      <td>NaN</td>\n",
       "      <td>227</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BMC Nephrol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>eng</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Place: England PMID: 31221135  PMCID: PMC6585015</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>*Blood Urea Nitrogen; Amino Acid Transport Systems, Basic/deficiency/genetics; Amino Acid Transport Systems, Neutral/deficiency/genetics; Animals; Chronic kidney disease; Cystine; Cystinuria; Cystinuria/*diagnostic imaging/genetics/*metabolism; Female; Kidney stones; Male; Mice; Mice, Knockout; Nephrolithiasis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>metabolic consequences of cystinuria. background: cystinuria is an inherited disorder of renal amino acid transport that causes recurrent nephrolithiasis and significant morbidity in humans. it has  an incidence of 1 in 7000 worldwide making it one of the most common genetic  disorders in man. we phenotypically characterized a mouse model of cystinuria  type a resultant from knockout of slc3a1. methods: knockout of slc3a1 at rna and  protein levels was evaluated using real-time quantitative pcr and  immunofluorescence. slc3a1 knockout mice were placed on normal or breeder chow  diets and evaluated for cystine stone formation over time suing x-ray analysis,  and the development of kidney injury by measuring injury biomarkers. kidney  injury was also evaluated via histologic analysis. amino acid levels were  measured in the blood of mice using high performance liquid chromatography. liver  glutathione levels were measured using a luminescent-based assay. results: we  confirmed knockout of slc3a1 at the rna level, while slc7a9 rna representing the  co-transporter was preserved. as expected, we observed bladder stone formation in  slc3a1(-/-) mice. male slc3a1(-/-) mice exhibited lower weights compared to  slc3a1(+/+). slc3a1(-/-) mice on a regular diet demonstrated elevated blood urea  nitrogen (bun) without elevation of serum creatinine. however, placing the  knockout animals on a breeder chow diet, containing a higher cystine  concentration, resulted in the development of elevation of both bun and  creatinine indicative of more severe chronic kidney disease. histological  examination revealed that these dietary effects resulted in worsened kidney  tubular obstruction and interstitial inflammation as well as worsened bladder  inflammation. cystine is a precursor for the antioxidant molecule glutathione, so  we evaluated glutathione levels in the livers of slc3a1(-/-) mice. we found  significantly lowered levels of both reduced and total glutathione in the  knockout animals. conclusions: our results suggest that that diet can affect the  development and progression of chronic kidney disease in an animal model of  cystinuria, which may have important implications for patients with this disease.  additionally, reduced glutathione may predispose those with cystinuria to injury  caused by oxidative stress. word count: 327.</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>STQLMMUE</td>\n",
       "      <td>journalArticle</td>\n",
       "      <td>2017</td>\n",
       "      <td>Song, Xuewen; Haghighi, Amirreza; Iliuta, Ioan-Andrei; Pei, York</td>\n",
       "      <td>Molecular diagnosis of autosomal dominant polycystic kidney disease.</td>\n",
       "      <td>Expert review of molecular diagnostics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1744-8352 1473-7159</td>\n",
       "      <td>10.1080/14737159.2017.1358088</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Autosomal dominant polycystic kidney disease (ADPKD) is the most common inherited kidney disease that accounts for 5-10% of end-stage renal disease in developed  countries. Mutations in PKD1 and PKD2 account for a majority of cases. Mutation  screening of PKD1 is technically challenging largely due to the complexity  resulting from duplication of its first 33 exons in six highly homologous  pseudogenes (i.e. PKD1P1-P6). Protocol using locus-specific long-range and nested  PCR has enabled comprehensive PKD1 mutation screening but is labor-intensive and  costly. Here, the authors review how recent advances in Next Generation  Sequencing are poised to transform and extend molecular diagnosis of ADPKD. Areas  covered: Key original research articles and reviews of the topic published in  English identified through PubMed from 1957-2017. Expert commentary: The authors  review current and evolving approaches using targeted resequencing or whole  genome sequencing for screening typical as well as challenging cases (e.g. cases  with no detectable PKD1 and PKD2 mutations which may be due to somatic mosaicism  or other cystic disease; and complex genetics such as bilineal disease).</td>\n",
       "      <td>2017-10</td>\n",
       "      <td>2025-07-29 13:45:51</td>\n",
       "      <td>2025-07-29 13:45:51</td>\n",
       "      <td>NaN</td>\n",
       "      <td>885-895</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Expert Rev Mol Diagn</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>eng</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Place: England PMID: 28724316</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>*Genetic Association Studies; *Genetic Testing/methods; Autosomal dominant polycystic kidney disease; Diagnosis, Differential; Genetic Markers; Genomics/methods; High-Throughput Nucleotide Sequencing; Humans; molecular diagnosis; Mutation; Next Generation Sequencing; PKD1; PKD2; Polycystic Kidney, Autosomal Dominant/*diagnosis/*genetics/metabolism/mortality; Prognosis; TRPP Cation Channels/genetics/metabolism</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>molecular diagnosis of autosomal dominant polycystic kidney disease. autosomal dominant polycystic kidney disease (adpkd) is the most common inherited kidney disease that accounts for 5-10% of end-stage renal disease in developed  countries. mutations in pkd1 and pkd2 account for a majority of cases. mutation  screening of pkd1 is technically challenging largely due to the complexity  resulting from duplication of its first 33 exons in six highly homologous  pseudogenes (i.e. pkd1p1-p6). protocol using locus-specific long-range and nested  pcr has enabled comprehensive pkd1 mutation screening but is labor-intensive and  costly. here, the authors review how recent advances in next generation  sequencing are poised to transform and extend molecular diagnosis of adpkd. areas  covered: key original research articles and reviews of the topic published in  english identified through pubmed from 1957-2017. expert commentary: the authors  review current and evolving approaches using targeted resequencing or whole  genome sequencing for screening typical as well as challenging cases (e.g. cases  with no detectable pkd1 and pkd2 mutations which may be due to somatic mosaicism  or other cystic disease; and complex genetics such as bilineal disease).</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Key       Item Type  Publication Year  \\\n",
       "0  REREKIS5  journalArticle              2021   \n",
       "1  AS467S2X  journalArticle              2019   \n",
       "2  6LN8B6N4  journalArticle              2019   \n",
       "3  5S9M622R  journalArticle              2019   \n",
       "4  STQLMMUE  journalArticle              2017   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                     Author  \\\n",
       "0  Stanzick, Kira J.; Li, Yong; Schlosser, Pascal; Gorski, Mathias; Wuttke, Matthias; Thomas, Laurent F.; Rasheed, Humaira; Rowan, Bryce X.; Graham, Sarah E.; Vanderweff, Brett R.; Patil, Snehal B.; Robinson-Cohen, Cassiane; Gaziano, John M.; O'Donnell, Christopher J.; Willer, Cristen J.; Hallan, Stein; Åsvold, Bjørn Olav; Gessner, Andre; Hung, Adriana M.; Pattaro, Cristian; Köttgen, Anna; Stark, Klaus J.; Heid, Iris M.; Winkler, Thomas W.   \n",
       "1                                                                                                                                                                                                                                                                                                                                                                                                                          George, Jaya A.; Gounden, Verena   \n",
       "2                                                                                                                                                                                                                                                                                                                                                        Capolongo, Giovanna; Zacchia, Miriam; Perna, Alessandra; Viggiano, Davide; Capasso, Giovambattista   \n",
       "3                                                                                                                                                                                                                                                                   Woodard, Lauren E.; Welch, Richard C.; Veach, Ruth Ann; Beckermann, Thomas M.; Sha, Feng; Weinman, Edward J.; Ikizler, Talat Alp; Tischfield, Jay A.; Sahota, Amrik; Wilson, Matthew H.   \n",
       "4                                                                                                                                                                                                                                                                                                                                                                                          Song, Xuewen; Haghighi, Amirreza; Iliuta, Ioan-Andrei; Pei, York   \n",
       "\n",
       "                                                                                                 Title  \\\n",
       "0  Discovery and prioritization of variants and genes for kidney function in >1.2 million individuals.   \n",
       "1                                                                 Novel glomerular filtration markers.   \n",
       "2                                                       Urinary proteome in inherited nephrolithiasis.   \n",
       "3                                                                Metabolic consequences of cystinuria.   \n",
       "4                                 Molecular diagnosis of autosomal dominant polycystic kidney disease.   \n",
       "\n",
       "                        Publication Title ISBN                 ISSN  \\\n",
       "0                   Nature communications  NaN            2041-1723   \n",
       "1          Advances in clinical chemistry  NaN            0065-2423   \n",
       "2                            Urolithiasis  NaN  2194-7236 2194-7228   \n",
       "3                          BMC nephrology  NaN            1471-2369   \n",
       "4  Expert review of molecular diagnostics  NaN  1744-8352 1473-7159   \n",
       "\n",
       "                             DOI  Url  \\\n",
       "0     10.1038/s41467-021-24491-0  NaN   \n",
       "1     10.1016/bs.acc.2018.10.005  NaN   \n",
       "2     10.1007/s00240-018-01104-y  NaN   \n",
       "3      10.1186/s12882-019-1417-8  NaN   \n",
       "4  10.1080/14737159.2017.1358088  NaN   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Abstract Note  \\\n",
       "0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Genes underneath signals from genome-wide association studies (GWAS) for kidney function are promising targets for functional studies, but prioritizing variants  and genes is challenging. By GWAS meta-analysis for creatinine-based estimated  glomerular filtration rate (eGFR) from the Chronic Kidney Disease Genetics  Consortium and UK Biobank (n = 1,201,909), we expand the number of eGFRcrea loci  (424 loci, 201 novel; 9.8% eGFRcrea variance explained by 634 independent signal  variants). Our increased sample size in fine-mapping (n = 1,004,040, European)  more than doubles the number of signals with resolved fine-mapping (99% credible  sets down to 1 variant for 44 signals, ≤5 variants for 138 signals).  Cystatin-based eGFR and/or blood urea nitrogen association support 348 loci  (n = 460,826 and 852,678, respectively). Our customizable tool for Gene  PrioritiSation reveals 23 compelling genes including mechanistic insights and  enables navigation through genes and variants likely relevant for kidney function  in human to help select targets for experimental follow-up.   \n",
       "1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Chronic kidney disease is currently assessed by estimated glomerular filtration rate, a mathematical construct based on creatinine or creatinine and cystatin  concentration. Creatinine-based equations have improved with standardization  efforts and the Modification of Diet in Renal Disease Study (MDRD) and  CKD-Epidemiology Collaboration Study (CKD-EPI). Because the measurement of  creatinine is subject to interference from non-GFR determinants, alternative  markers have long been sought. These have included cystatin C and low molecular  weight proteins like β2-microglobulin and beta trace protein. Tubular disease  often occurs before glomerular filtration is impaired and investigators have  investigated the excretion of other low molecular weight proteins such as  Neutrophil Gelatinase-Associated Lipocalin (NGAL) and Kidney Injury Molecule-1  and N-acetyl-β-d-glucosaminidase. While preliminary, there is some evidence  linking these analytes with GFR, disease stage and mortality. Although  asymmetrical dimethyl arginine, an inhibitor of nitric oxide, has been shown to  be associated with progression of renal disease, symmetric dimethyl arginine may  be a better marker. Recent work has also explored the potential of microRNA  (miRNA) analysis and metabolomics studies to further elucidate this complex  pathophysiologic disease process. Investigators hope to improve our ability to  detect CKD by the use of test panels, i.e., various marker combinations thereof.  Unfortunately, most of these markers lack standardization unlike traditional  measures that rely on creatinine and cystatin C measurement.   \n",
       "2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              In the last decades, proteomics has been largely applied to the Nephrology field, with the double aim to (1) elucidate the biological processes underlying renal  diseases; (2) identify disease-specific biomarkers, predictor factors of  therapeutic efficacy and prognostic factors of disease progression. Kidney stone  disease, and in particular, inherited nephrolithiasis (INL) are not an exception.  Given the multifactorial origin of these disorders, the combination of genomics  and proteomics studies may complement each other, with the final objective to  give a global and comprehensive mechanistic view. In this review, we summarize  the results of recent proteomic studies which have expanded our knowledge about  INL, focusing the attention on monogenic forms of nephrolithiasis (cystinuria,  Dent's disease, Bartter syndrome, distal renal tubular acidosis and primary  hyperoxaluria), on polygenic hypercalciuria and on medullary sponge kidney  disease.   \n",
       "3  BACKGROUND: Cystinuria is an inherited disorder of renal amino acid transport that causes recurrent nephrolithiasis and significant morbidity in humans. It has  an incidence of 1 in 7000 worldwide making it one of the most common genetic  disorders in man. We phenotypically characterized a mouse model of cystinuria  type A resultant from knockout of Slc3a1. METHODS: Knockout of Slc3a1 at RNA and  protein levels was evaluated using real-time quantitative PCR and  immunofluorescence. Slc3a1 knockout mice were placed on normal or breeder chow  diets and evaluated for cystine stone formation over time suing x-ray analysis,  and the development of kidney injury by measuring injury biomarkers. Kidney  injury was also evaluated via histologic analysis. Amino acid levels were  measured in the blood of mice using high performance liquid chromatography. Liver  glutathione levels were measured using a luminescent-based assay. RESULTS: We  confirmed knockout of Slc3a1 at the RNA level, while Slc7a9 RNA representing the  co-transporter was preserved. As expected, we observed bladder stone formation in  Slc3a1(-/-) mice. Male Slc3a1(-/-) mice exhibited lower weights compared to  Slc3a1(+/+). Slc3a1(-/-) mice on a regular diet demonstrated elevated blood urea  nitrogen (BUN) without elevation of serum creatinine. However, placing the  knockout animals on a breeder chow diet, containing a higher cystine  concentration, resulted in the development of elevation of both BUN and  creatinine indicative of more severe chronic kidney disease. Histological  examination revealed that these dietary effects resulted in worsened kidney  tubular obstruction and interstitial inflammation as well as worsened bladder  inflammation. Cystine is a precursor for the antioxidant molecule glutathione, so  we evaluated glutathione levels in the livers of Slc3a1(-/-) mice. We found  significantly lowered levels of both reduced and total glutathione in the  knockout animals. CONCLUSIONS: Our results suggest that that diet can affect the  development and progression of chronic kidney disease in an animal model of  cystinuria, which may have important implications for patients with this disease.  Additionally, reduced glutathione may predispose those with cystinuria to injury  caused by oxidative stress. Word count: 327.   \n",
       "4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Autosomal dominant polycystic kidney disease (ADPKD) is the most common inherited kidney disease that accounts for 5-10% of end-stage renal disease in developed  countries. Mutations in PKD1 and PKD2 account for a majority of cases. Mutation  screening of PKD1 is technically challenging largely due to the complexity  resulting from duplication of its first 33 exons in six highly homologous  pseudogenes (i.e. PKD1P1-P6). Protocol using locus-specific long-range and nested  PCR has enabled comprehensive PKD1 mutation screening but is labor-intensive and  costly. Here, the authors review how recent advances in Next Generation  Sequencing are poised to transform and extend molecular diagnosis of ADPKD. Areas  covered: Key original research articles and reviews of the topic published in  English identified through PubMed from 1957-2017. Expert commentary: The authors  review current and evolving approaches using targeted resequencing or whole  genome sequencing for screening typical as well as challenging cases (e.g. cases  with no detectable PKD1 and PKD2 mutations which may be due to somatic mosaicism  or other cystic disease; and complex genetics such as bilineal disease).   \n",
       "\n",
       "         Date           Date Added        Date Modified Access Date    Pages  \\\n",
       "0  2021-07-16  2025-07-29 13:45:51  2025-07-29 13:45:51         NaN     4350   \n",
       "1        2019  2025-07-29 13:45:51  2025-07-29 13:45:51         NaN   91-119   \n",
       "2     2019-02  2025-07-29 13:45:51  2025-07-29 13:45:51         NaN    91-98   \n",
       "3  2019-06-20  2025-07-29 13:45:51  2025-07-29 13:45:51         NaN      227   \n",
       "4     2017-10  2025-07-29 13:45:51  2025-07-29 13:45:51         NaN  885-895   \n",
       "\n",
       "   Num Pages Issue Volume  Number Of Volumes  Journal Abbreviation  \\\n",
       "0        NaN     1     12                NaN            Nat Commun   \n",
       "1        NaN   NaN     88                NaN         Adv Clin Chem   \n",
       "2        NaN     1     47                NaN          Urolithiasis   \n",
       "3        NaN     1     20                NaN           BMC Nephrol   \n",
       "4        NaN    10     17                NaN  Expert Rev Mol Diagn   \n",
       "\n",
       "   Short Title  Series  Series Number  Series Text  Series Title  Publisher  \\\n",
       "0          NaN     NaN            NaN          NaN           NaN        NaN   \n",
       "1          NaN     NaN            NaN          NaN           NaN        NaN   \n",
       "2          NaN     NaN            NaN          NaN           NaN        NaN   \n",
       "3          NaN     NaN            NaN          NaN           NaN        NaN   \n",
       "4          NaN     NaN            NaN          NaN           NaN        NaN   \n",
       "\n",
       "   Place Language                                     Rights  Type  Archive  \\\n",
       "0    NaN      eng                     © 2021. The Author(s).   NaN      NaN   \n",
       "1    NaN      eng  © 2019 Elsevier Inc. All rights reserved.   NaN      NaN   \n",
       "2    NaN      eng                                        NaN   NaN      NaN   \n",
       "3    NaN      eng                                        NaN   NaN      NaN   \n",
       "4    NaN      eng                                        NaN   NaN      NaN   \n",
       "\n",
       "  Archive Location  Library Catalog  Call Number  \\\n",
       "0              NaN              NaN          NaN   \n",
       "1              NaN              NaN          NaN   \n",
       "2              NaN              NaN          NaN   \n",
       "3              NaN              NaN          NaN   \n",
       "4              NaN              NaN          NaN   \n",
       "\n",
       "                                              Extra  Notes  File Attachments  \\\n",
       "0  Place: England PMID: 34272381  PMCID: PMC8285412    NaN               NaN   \n",
       "1               Place: United States PMID: 30612608    NaN               NaN   \n",
       "2                     Place: Germany PMID: 30564846    NaN               NaN   \n",
       "3  Place: England PMID: 31221135  PMCID: PMC6585015    NaN               NaN   \n",
       "4                     Place: England PMID: 28724316    NaN               NaN   \n",
       "\n",
       "   Link Attachments  \\\n",
       "0               NaN   \n",
       "1               NaN   \n",
       "2               NaN   \n",
       "3               NaN   \n",
       "4               NaN   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                        Manual Tags  \\\n",
       "0                 *Genetic Predisposition to Disease; Biomarkers; Creatinine/blood; Cystatins/pharmacology; Databases, Genetic; Europe; Gene Expression Regulation/genetics; Genome-Wide Association Study; Glomerular Filtration Rate/*genetics; Humans; Kidney/*metabolism/physiology; Organ Specificity; Quantitative Trait Loci; Renal Insufficiency, Chronic/genetics/*metabolism; Risk Factors; RNA-Seq; Single-Cell Analysis   \n",
       "1  Acetylglucosaminidase/analysis; Animals; Arginine/analogs & derivatives/analysis; beta 2-Microglobulin/analysis; Biomarkers/analysis; Creatinine/analysis; Cystatin C; Cystatin C/analysis; GFR; Glomerular Filtration Rate; Hepatitis A Virus Cellular Receptor 1/analysis; Humans; Kidney/physiopathology; KIM-1; Lipocalin-2/analysis; miRNA; NGAL; Renal Insufficiency, Chronic/*diagnosis/physiopathology; β2-microglobulin   \n",
       "2                                                                                                                                                                  Biomarker; Biomarkers/urine; Genetic nephrolithiasis; Humans; Hypercalciuria/diagnosis/genetics/*urine; Medullary Sponge Kidney/diagnosis/genetics/*urine; Nephrolithiasis/diagnosis/genetics/*urine; Proteome/*analysis; Proteomics; Proteomics/*methods; Urine   \n",
       "3                                                                                                           *Blood Urea Nitrogen; Amino Acid Transport Systems, Basic/deficiency/genetics; Amino Acid Transport Systems, Neutral/deficiency/genetics; Animals; Chronic kidney disease; Cystine; Cystinuria; Cystinuria/*diagnostic imaging/genetics/*metabolism; Female; Kidney stones; Male; Mice; Mice, Knockout; Nephrolithiasis   \n",
       "4      *Genetic Association Studies; *Genetic Testing/methods; Autosomal dominant polycystic kidney disease; Diagnosis, Differential; Genetic Markers; Genomics/methods; High-Throughput Nucleotide Sequencing; Humans; molecular diagnosis; Mutation; Next Generation Sequencing; PKD1; PKD2; Polycystic Kidney, Autosomal Dominant/*diagnosis/*genetics/metabolism/mortality; Prognosis; TRPP Cation Channels/genetics/metabolism   \n",
       "\n",
       "   Automatic Tags Editor  Series Editor  Translator  Contributor  \\\n",
       "0             NaN    NaN            NaN         NaN          NaN   \n",
       "1             NaN    NaN            NaN         NaN          NaN   \n",
       "2             NaN    NaN            NaN         NaN          NaN   \n",
       "3             NaN    NaN            NaN         NaN          NaN   \n",
       "4             NaN    NaN            NaN         NaN          NaN   \n",
       "\n",
       "   Attorney Agent  Book Author  Cast Member  Commenter  Composer  Cosponsor  \\\n",
       "0             NaN          NaN          NaN        NaN       NaN        NaN   \n",
       "1             NaN          NaN          NaN        NaN       NaN        NaN   \n",
       "2             NaN          NaN          NaN        NaN       NaN        NaN   \n",
       "3             NaN          NaN          NaN        NaN       NaN        NaN   \n",
       "4             NaN          NaN          NaN        NaN       NaN        NaN   \n",
       "\n",
       "   Counsel  Interviewer  Producer  Recipient  Reviewed Author  Scriptwriter  \\\n",
       "0      NaN          NaN       NaN        NaN              NaN           NaN   \n",
       "1      NaN          NaN       NaN        NaN              NaN           NaN   \n",
       "2      NaN          NaN       NaN        NaN              NaN           NaN   \n",
       "3      NaN          NaN       NaN        NaN              NaN           NaN   \n",
       "4      NaN          NaN       NaN        NaN              NaN           NaN   \n",
       "\n",
       "   Words By  Guest  Number  Edition  Running Time  Scale  Medium  \\\n",
       "0       NaN    NaN     NaN      NaN           NaN    NaN     NaN   \n",
       "1       NaN    NaN     NaN      NaN           NaN    NaN     NaN   \n",
       "2       NaN    NaN     NaN      NaN           NaN    NaN     NaN   \n",
       "3       NaN    NaN     NaN      NaN           NaN    NaN     NaN   \n",
       "4       NaN    NaN     NaN      NaN           NaN    NaN     NaN   \n",
       "\n",
       "   Artwork Size  Filing Date  Application Number  Assignee  Issuing Authority  \\\n",
       "0           NaN          NaN                 NaN       NaN                NaN   \n",
       "1           NaN          NaN                 NaN       NaN                NaN   \n",
       "2           NaN          NaN                 NaN       NaN                NaN   \n",
       "3           NaN          NaN                 NaN       NaN                NaN   \n",
       "4           NaN          NaN                 NaN       NaN                NaN   \n",
       "\n",
       "   Country  Meeting Name Conference Name  Court  References  Reporter  \\\n",
       "0      NaN           NaN             NaN    NaN         NaN       NaN   \n",
       "1      NaN           NaN             NaN    NaN         NaN       NaN   \n",
       "2      NaN           NaN             NaN    NaN         NaN       NaN   \n",
       "3      NaN           NaN             NaN    NaN         NaN       NaN   \n",
       "4      NaN           NaN             NaN    NaN         NaN       NaN   \n",
       "\n",
       "   Legal Status  Priority Numbers  Programming Language  Version  System  \\\n",
       "0           NaN               NaN                   NaN      NaN     NaN   \n",
       "1           NaN               NaN                   NaN      NaN     NaN   \n",
       "2           NaN               NaN                   NaN      NaN     NaN   \n",
       "3           NaN               NaN                   NaN      NaN     NaN   \n",
       "4           NaN               NaN                   NaN      NaN     NaN   \n",
       "\n",
       "   Code  Code Number  Section  Session  Committee  History  Legislative Body  \\\n",
       "0   NaN          NaN      NaN      NaN        NaN      NaN               NaN   \n",
       "1   NaN          NaN      NaN      NaN        NaN      NaN               NaN   \n",
       "2   NaN          NaN      NaN      NaN        NaN      NaN               NaN   \n",
       "3   NaN          NaN      NaN      NaN        NaN      NaN               NaN   \n",
       "4   NaN          NaN      NaN      NaN        NaN      NaN               NaN   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           combined_text  \\\n",
       "0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      discovery and prioritization of variants and genes for kidney function in >1.2 million individuals. genes underneath signals from genome-wide association studies (gwas) for kidney function are promising targets for functional studies, but prioritizing variants  and genes is challenging. by gwas meta-analysis for creatinine-based estimated  glomerular filtration rate (egfr) from the chronic kidney disease genetics  consortium and uk biobank (n = 1,201,909), we expand the number of egfrcrea loci  (424 loci, 201 novel; 9.8% egfrcrea variance explained by 634 independent signal  variants). our increased sample size in fine-mapping (n = 1,004,040, european)  more than doubles the number of signals with resolved fine-mapping (99% credible  sets down to 1 variant for 44 signals, ≤5 variants for 138 signals).  cystatin-based egfr and/or blood urea nitrogen association support 348 loci  (n = 460,826 and 852,678, respectively). our customizable tool for gene  prioritisation reveals 23 compelling genes including mechanistic insights and  enables navigation through genes and variants likely relevant for kidney function  in human to help select targets for experimental follow-up.   \n",
       "1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            novel glomerular filtration markers. chronic kidney disease is currently assessed by estimated glomerular filtration rate, a mathematical construct based on creatinine or creatinine and cystatin  concentration. creatinine-based equations have improved with standardization  efforts and the modification of diet in renal disease study (mdrd) and  ckd-epidemiology collaboration study (ckd-epi). because the measurement of  creatinine is subject to interference from non-gfr determinants, alternative  markers have long been sought. these have included cystatin c and low molecular  weight proteins like β2-microglobulin and beta trace protein. tubular disease  often occurs before glomerular filtration is impaired and investigators have  investigated the excretion of other low molecular weight proteins such as  neutrophil gelatinase-associated lipocalin (ngal) and kidney injury molecule-1  and n-acetyl-β-d-glucosaminidase. while preliminary, there is some evidence  linking these analytes with gfr, disease stage and mortality. although  asymmetrical dimethyl arginine, an inhibitor of nitric oxide, has been shown to  be associated with progression of renal disease, symmetric dimethyl arginine may  be a better marker. recent work has also explored the potential of microrna  (mirna) analysis and metabolomics studies to further elucidate this complex  pathophysiologic disease process. investigators hope to improve our ability to  detect ckd by the use of test panels, i.e., various marker combinations thereof.  unfortunately, most of these markers lack standardization unlike traditional  measures that rely on creatinine and cystatin c measurement.   \n",
       "2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     urinary proteome in inherited nephrolithiasis. in the last decades, proteomics has been largely applied to the nephrology field, with the double aim to (1) elucidate the biological processes underlying renal  diseases; (2) identify disease-specific biomarkers, predictor factors of  therapeutic efficacy and prognostic factors of disease progression. kidney stone  disease, and in particular, inherited nephrolithiasis (inl) are not an exception.  given the multifactorial origin of these disorders, the combination of genomics  and proteomics studies may complement each other, with the final objective to  give a global and comprehensive mechanistic view. in this review, we summarize  the results of recent proteomic studies which have expanded our knowledge about  inl, focusing the attention on monogenic forms of nephrolithiasis (cystinuria,  dent's disease, bartter syndrome, distal renal tubular acidosis and primary  hyperoxaluria), on polygenic hypercalciuria and on medullary sponge kidney  disease.   \n",
       "3  metabolic consequences of cystinuria. background: cystinuria is an inherited disorder of renal amino acid transport that causes recurrent nephrolithiasis and significant morbidity in humans. it has  an incidence of 1 in 7000 worldwide making it one of the most common genetic  disorders in man. we phenotypically characterized a mouse model of cystinuria  type a resultant from knockout of slc3a1. methods: knockout of slc3a1 at rna and  protein levels was evaluated using real-time quantitative pcr and  immunofluorescence. slc3a1 knockout mice were placed on normal or breeder chow  diets and evaluated for cystine stone formation over time suing x-ray analysis,  and the development of kidney injury by measuring injury biomarkers. kidney  injury was also evaluated via histologic analysis. amino acid levels were  measured in the blood of mice using high performance liquid chromatography. liver  glutathione levels were measured using a luminescent-based assay. results: we  confirmed knockout of slc3a1 at the rna level, while slc7a9 rna representing the  co-transporter was preserved. as expected, we observed bladder stone formation in  slc3a1(-/-) mice. male slc3a1(-/-) mice exhibited lower weights compared to  slc3a1(+/+). slc3a1(-/-) mice on a regular diet demonstrated elevated blood urea  nitrogen (bun) without elevation of serum creatinine. however, placing the  knockout animals on a breeder chow diet, containing a higher cystine  concentration, resulted in the development of elevation of both bun and  creatinine indicative of more severe chronic kidney disease. histological  examination revealed that these dietary effects resulted in worsened kidney  tubular obstruction and interstitial inflammation as well as worsened bladder  inflammation. cystine is a precursor for the antioxidant molecule glutathione, so  we evaluated glutathione levels in the livers of slc3a1(-/-) mice. we found  significantly lowered levels of both reduced and total glutathione in the  knockout animals. conclusions: our results suggest that that diet can affect the  development and progression of chronic kidney disease in an animal model of  cystinuria, which may have important implications for patients with this disease.  additionally, reduced glutathione may predispose those with cystinuria to injury  caused by oxidative stress. word count: 327.   \n",
       "4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             molecular diagnosis of autosomal dominant polycystic kidney disease. autosomal dominant polycystic kidney disease (adpkd) is the most common inherited kidney disease that accounts for 5-10% of end-stage renal disease in developed  countries. mutations in pkd1 and pkd2 account for a majority of cases. mutation  screening of pkd1 is technically challenging largely due to the complexity  resulting from duplication of its first 33 exons in six highly homologous  pseudogenes (i.e. pkd1p1-p6). protocol using locus-specific long-range and nested  pcr has enabled comprehensive pkd1 mutation screening but is labor-intensive and  costly. here, the authors review how recent advances in next generation  sequencing are poised to transform and extend molecular diagnosis of adpkd. areas  covered: key original research articles and reviews of the topic published in  english identified through pubmed from 1957-2017. expert commentary: the authors  review current and evolving approaches using targeted resequencing or whole  genome sequencing for screening typical as well as challenging cases (e.g. cases  with no detectable pkd1 and pkd2 mutations which may be due to somatic mosaicism  or other cystic disease; and complex genetics such as bilineal disease).   \n",
       "\n",
       "  decision  has_abstract  \n",
       "0     None          True  \n",
       "1     None          True  \n",
       "2     None          True  \n",
       "3     None          True  \n",
       "4     None          True  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# como no hemos recuperado ningún abstract, podemos reasignar df y exportar\n",
    "df = df[df[\"has_abstract\"]==True]\n",
    "print(f\"Shape: {df.shape}\")\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "21864472-5357-47db-943b-23ad60cbc894",
   "metadata": {},
   "source": [
    "# 3.revisar manualmente abstracts no recuperados"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "5a87bf62-2064-4886-9d67-9765bf122118",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Key</th>\n",
       "      <th>Item Type</th>\n",
       "      <th>Publication Year</th>\n",
       "      <th>Author</th>\n",
       "      <th>Title</th>\n",
       "      <th>Publication Title</th>\n",
       "      <th>ISBN</th>\n",
       "      <th>ISSN</th>\n",
       "      <th>DOI</th>\n",
       "      <th>Url</th>\n",
       "      <th>Abstract Note</th>\n",
       "      <th>Date</th>\n",
       "      <th>Date Added</th>\n",
       "      <th>Date Modified</th>\n",
       "      <th>Access Date</th>\n",
       "      <th>Pages</th>\n",
       "      <th>Num Pages</th>\n",
       "      <th>Issue</th>\n",
       "      <th>Volume</th>\n",
       "      <th>Number Of Volumes</th>\n",
       "      <th>Journal Abbreviation</th>\n",
       "      <th>Short Title</th>\n",
       "      <th>Series</th>\n",
       "      <th>Series Number</th>\n",
       "      <th>Series Text</th>\n",
       "      <th>Series Title</th>\n",
       "      <th>Publisher</th>\n",
       "      <th>Place</th>\n",
       "      <th>Language</th>\n",
       "      <th>Rights</th>\n",
       "      <th>Type</th>\n",
       "      <th>Archive</th>\n",
       "      <th>Archive Location</th>\n",
       "      <th>Library Catalog</th>\n",
       "      <th>Call Number</th>\n",
       "      <th>Extra</th>\n",
       "      <th>Notes</th>\n",
       "      <th>File Attachments</th>\n",
       "      <th>Link Attachments</th>\n",
       "      <th>Manual Tags</th>\n",
       "      <th>Automatic Tags</th>\n",
       "      <th>Editor</th>\n",
       "      <th>Series Editor</th>\n",
       "      <th>Translator</th>\n",
       "      <th>Contributor</th>\n",
       "      <th>Attorney Agent</th>\n",
       "      <th>Book Author</th>\n",
       "      <th>Cast Member</th>\n",
       "      <th>Commenter</th>\n",
       "      <th>Composer</th>\n",
       "      <th>Cosponsor</th>\n",
       "      <th>Counsel</th>\n",
       "      <th>Interviewer</th>\n",
       "      <th>Producer</th>\n",
       "      <th>Recipient</th>\n",
       "      <th>Reviewed Author</th>\n",
       "      <th>Scriptwriter</th>\n",
       "      <th>Words By</th>\n",
       "      <th>Guest</th>\n",
       "      <th>Number</th>\n",
       "      <th>Edition</th>\n",
       "      <th>Running Time</th>\n",
       "      <th>Scale</th>\n",
       "      <th>Medium</th>\n",
       "      <th>Artwork Size</th>\n",
       "      <th>Filing Date</th>\n",
       "      <th>Application Number</th>\n",
       "      <th>Assignee</th>\n",
       "      <th>Issuing Authority</th>\n",
       "      <th>Country</th>\n",
       "      <th>Meeting Name</th>\n",
       "      <th>Conference Name</th>\n",
       "      <th>Court</th>\n",
       "      <th>References</th>\n",
       "      <th>Reporter</th>\n",
       "      <th>Legal Status</th>\n",
       "      <th>Priority Numbers</th>\n",
       "      <th>Programming Language</th>\n",
       "      <th>Version</th>\n",
       "      <th>System</th>\n",
       "      <th>Code</th>\n",
       "      <th>Code Number</th>\n",
       "      <th>Section</th>\n",
       "      <th>Session</th>\n",
       "      <th>Committee</th>\n",
       "      <th>History</th>\n",
       "      <th>Legislative Body</th>\n",
       "      <th>combined_text</th>\n",
       "      <th>decision</th>\n",
       "      <th>has_abstract</th>\n",
       "      <th>ADPKD?</th>\n",
       "      <th>OMICS?</th>\n",
       "      <th>BIOINFO?</th>\n",
       "      <th>HUMAN DATA?</th>\n",
       "      <th>BIOMARKER?</th>\n",
       "      <th>TRASLATIONAL?</th>\n",
       "      <th>DECISION</th>\n",
       "      <th>COMMENT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>REREKIS5</td>\n",
       "      <td>journalArticle</td>\n",
       "      <td>2021</td>\n",
       "      <td>Stanzick, Kira J.; Li, Yong; Schlosser, Pascal; Gorski, Mathias; Wuttke, Matthias; Thomas, Laurent F.; Rasheed, Humaira; Rowan, Bryce X.; Graham, Sarah E.; Vanderweff, Brett R.; Patil, Snehal B.; Robinson-Cohen, Cassiane; Gaziano, John M.; O'Donnell, Christopher J.; Willer, Cristen J.; Hallan, Stein; Åsvold, Bjørn Olav; Gessner, Andre; Hung, Adriana M.; Pattaro, Cristian; Köttgen, Anna; Stark, Klaus J.; Heid, Iris M.; Winkler, Thomas W.</td>\n",
       "      <td>Discovery and prioritization of variants and genes for kidney function in &gt;1.2 million individuals.</td>\n",
       "      <td>Nature communications</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2041-1723</td>\n",
       "      <td>10.1038/s41467-021-24491-0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Genes underneath signals from genome-wide association studies (GWAS) for kidney function are promising targets for functional studies, but prioritizing variants  and genes is challenging. By GWAS meta-analysis for creatinine-based estimated  glomerular filtration rate (eGFR) from the Chronic Kidney Disease Genetics  Consortium and UK Biobank (n = 1,201,909), we expand the number of eGFRcrea loci  (424 loci, 201 novel; 9.8% eGFRcrea variance explained by 634 independent signal  variants). Our increased sample size in fine-mapping (n = 1,004,040, European)  more than doubles the number of signals with resolved fine-mapping (99% credible  sets down to 1 variant for 44 signals, ≤5 variants for 138 signals).  Cystatin-based eGFR and/or blood urea nitrogen association support 348 loci  (n = 460,826 and 852,678, respectively). Our customizable tool for Gene  PrioritiSation reveals 23 compelling genes including mechanistic insights and  enables navigation through genes and variants likely relevant for kidney function  in human to help select targets for experimental follow-up.</td>\n",
       "      <td>2021-07-16</td>\n",
       "      <td>2025-07-29 13:45:51</td>\n",
       "      <td>2025-07-29 13:45:51</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4350</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nat Commun</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>eng</td>\n",
       "      <td>© 2021. The Author(s).</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Place: England PMID: 34272381  PMCID: PMC8285412</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>*Genetic Predisposition to Disease; Biomarkers; Creatinine/blood; Cystatins/pharmacology; Databases, Genetic; Europe; Gene Expression Regulation/genetics; Genome-Wide Association Study; Glomerular Filtration Rate/*genetics; Humans; Kidney/*metabolism/physiology; Organ Specificity; Quantitative Trait Loci; Renal Insufficiency, Chronic/genetics/*metabolism; Risk Factors; RNA-Seq; Single-Cell Analysis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>discovery and prioritization of variants and genes for kidney function in &gt;1.2 million individuals. genes underneath signals from genome-wide association studies (gwas) for kidney function are promising targets for functional studies, but prioritizing variants  and genes is challenging. by gwas meta-analysis for creatinine-based estimated  glomerular filtration rate (egfr) from the chronic kidney disease genetics  consortium and uk biobank (n = 1,201,909), we expand the number of egfrcrea loci  (424 loci, 201 novel; 9.8% egfrcrea variance explained by 634 independent signal  variants). our increased sample size in fine-mapping (n = 1,004,040, european)  more than doubles the number of signals with resolved fine-mapping (99% credible  sets down to 1 variant for 44 signals, ≤5 variants for 138 signals).  cystatin-based egfr and/or blood urea nitrogen association support 348 loci  (n = 460,826 and 852,678, respectively). our customizable tool for gene  prioritisation reveals 23 compelling genes including mechanistic insights and  enables navigation through genes and variants likely relevant for kidney function  in human to help select targets for experimental follow-up.</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AS467S2X</td>\n",
       "      <td>journalArticle</td>\n",
       "      <td>2019</td>\n",
       "      <td>George, Jaya A.; Gounden, Verena</td>\n",
       "      <td>Novel glomerular filtration markers.</td>\n",
       "      <td>Advances in clinical chemistry</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0065-2423</td>\n",
       "      <td>10.1016/bs.acc.2018.10.005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chronic kidney disease is currently assessed by estimated glomerular filtration rate, a mathematical construct based on creatinine or creatinine and cystatin  concentration. Creatinine-based equations have improved with standardization  efforts and the Modification of Diet in Renal Disease Study (MDRD) and  CKD-Epidemiology Collaboration Study (CKD-EPI). Because the measurement of  creatinine is subject to interference from non-GFR determinants, alternative  markers have long been sought. These have included cystatin C and low molecular  weight proteins like β2-microglobulin and beta trace protein. Tubular disease  often occurs before glomerular filtration is impaired and investigators have  investigated the excretion of other low molecular weight proteins such as  Neutrophil Gelatinase-Associated Lipocalin (NGAL) and Kidney Injury Molecule-1  and N-acetyl-β-d-glucosaminidase. While preliminary, there is some evidence  linking these analytes with GFR, disease stage and mortality. Although  asymmetrical dimethyl arginine, an inhibitor of nitric oxide, has been shown to  be associated with progression of renal disease, symmetric dimethyl arginine may  be a better marker. Recent work has also explored the potential of microRNA  (miRNA) analysis and metabolomics studies to further elucidate this complex  pathophysiologic disease process. Investigators hope to improve our ability to  detect CKD by the use of test panels, i.e., various marker combinations thereof.  Unfortunately, most of these markers lack standardization unlike traditional  measures that rely on creatinine and cystatin C measurement.</td>\n",
       "      <td>2019</td>\n",
       "      <td>2025-07-29 13:45:51</td>\n",
       "      <td>2025-07-29 13:45:51</td>\n",
       "      <td>NaN</td>\n",
       "      <td>91-119</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Adv Clin Chem</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>eng</td>\n",
       "      <td>© 2019 Elsevier Inc. All rights reserved.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Place: United States PMID: 30612608</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Acetylglucosaminidase/analysis; Animals; Arginine/analogs &amp; derivatives/analysis; beta 2-Microglobulin/analysis; Biomarkers/analysis; Creatinine/analysis; Cystatin C; Cystatin C/analysis; GFR; Glomerular Filtration Rate; Hepatitis A Virus Cellular Receptor 1/analysis; Humans; Kidney/physiopathology; KIM-1; Lipocalin-2/analysis; miRNA; NGAL; Renal Insufficiency, Chronic/*diagnosis/physiopathology; β2-microglobulin</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>novel glomerular filtration markers. chronic kidney disease is currently assessed by estimated glomerular filtration rate, a mathematical construct based on creatinine or creatinine and cystatin  concentration. creatinine-based equations have improved with standardization  efforts and the modification of diet in renal disease study (mdrd) and  ckd-epidemiology collaboration study (ckd-epi). because the measurement of  creatinine is subject to interference from non-gfr determinants, alternative  markers have long been sought. these have included cystatin c and low molecular  weight proteins like β2-microglobulin and beta trace protein. tubular disease  often occurs before glomerular filtration is impaired and investigators have  investigated the excretion of other low molecular weight proteins such as  neutrophil gelatinase-associated lipocalin (ngal) and kidney injury molecule-1  and n-acetyl-β-d-glucosaminidase. while preliminary, there is some evidence  linking these analytes with gfr, disease stage and mortality. although  asymmetrical dimethyl arginine, an inhibitor of nitric oxide, has been shown to  be associated with progression of renal disease, symmetric dimethyl arginine may  be a better marker. recent work has also explored the potential of microrna  (mirna) analysis and metabolomics studies to further elucidate this complex  pathophysiologic disease process. investigators hope to improve our ability to  detect ckd by the use of test panels, i.e., various marker combinations thereof.  unfortunately, most of these markers lack standardization unlike traditional  measures that rely on creatinine and cystatin c measurement.</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Key       Item Type  Publication Year  \\\n",
       "0  REREKIS5  journalArticle              2021   \n",
       "1  AS467S2X  journalArticle              2019   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                     Author  \\\n",
       "0  Stanzick, Kira J.; Li, Yong; Schlosser, Pascal; Gorski, Mathias; Wuttke, Matthias; Thomas, Laurent F.; Rasheed, Humaira; Rowan, Bryce X.; Graham, Sarah E.; Vanderweff, Brett R.; Patil, Snehal B.; Robinson-Cohen, Cassiane; Gaziano, John M.; O'Donnell, Christopher J.; Willer, Cristen J.; Hallan, Stein; Åsvold, Bjørn Olav; Gessner, Andre; Hung, Adriana M.; Pattaro, Cristian; Köttgen, Anna; Stark, Klaus J.; Heid, Iris M.; Winkler, Thomas W.   \n",
       "1                                                                                                                                                                                                                                                                                                                                                                                                                          George, Jaya A.; Gounden, Verena   \n",
       "\n",
       "                                                                                                 Title  \\\n",
       "0  Discovery and prioritization of variants and genes for kidney function in >1.2 million individuals.   \n",
       "1                                                                 Novel glomerular filtration markers.   \n",
       "\n",
       "                Publication Title ISBN       ISSN                         DOI  \\\n",
       "0           Nature communications  NaN  2041-1723  10.1038/s41467-021-24491-0   \n",
       "1  Advances in clinical chemistry  NaN  0065-2423  10.1016/bs.acc.2018.10.005   \n",
       "\n",
       "   Url  \\\n",
       "0  NaN   \n",
       "1  NaN   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Abstract Note  \\\n",
       "0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Genes underneath signals from genome-wide association studies (GWAS) for kidney function are promising targets for functional studies, but prioritizing variants  and genes is challenging. By GWAS meta-analysis for creatinine-based estimated  glomerular filtration rate (eGFR) from the Chronic Kidney Disease Genetics  Consortium and UK Biobank (n = 1,201,909), we expand the number of eGFRcrea loci  (424 loci, 201 novel; 9.8% eGFRcrea variance explained by 634 independent signal  variants). Our increased sample size in fine-mapping (n = 1,004,040, European)  more than doubles the number of signals with resolved fine-mapping (99% credible  sets down to 1 variant for 44 signals, ≤5 variants for 138 signals).  Cystatin-based eGFR and/or blood urea nitrogen association support 348 loci  (n = 460,826 and 852,678, respectively). Our customizable tool for Gene  PrioritiSation reveals 23 compelling genes including mechanistic insights and  enables navigation through genes and variants likely relevant for kidney function  in human to help select targets for experimental follow-up.   \n",
       "1  Chronic kidney disease is currently assessed by estimated glomerular filtration rate, a mathematical construct based on creatinine or creatinine and cystatin  concentration. Creatinine-based equations have improved with standardization  efforts and the Modification of Diet in Renal Disease Study (MDRD) and  CKD-Epidemiology Collaboration Study (CKD-EPI). Because the measurement of  creatinine is subject to interference from non-GFR determinants, alternative  markers have long been sought. These have included cystatin C and low molecular  weight proteins like β2-microglobulin and beta trace protein. Tubular disease  often occurs before glomerular filtration is impaired and investigators have  investigated the excretion of other low molecular weight proteins such as  Neutrophil Gelatinase-Associated Lipocalin (NGAL) and Kidney Injury Molecule-1  and N-acetyl-β-d-glucosaminidase. While preliminary, there is some evidence  linking these analytes with GFR, disease stage and mortality. Although  asymmetrical dimethyl arginine, an inhibitor of nitric oxide, has been shown to  be associated with progression of renal disease, symmetric dimethyl arginine may  be a better marker. Recent work has also explored the potential of microRNA  (miRNA) analysis and metabolomics studies to further elucidate this complex  pathophysiologic disease process. Investigators hope to improve our ability to  detect CKD by the use of test panels, i.e., various marker combinations thereof.  Unfortunately, most of these markers lack standardization unlike traditional  measures that rely on creatinine and cystatin C measurement.   \n",
       "\n",
       "         Date           Date Added        Date Modified Access Date   Pages  \\\n",
       "0  2021-07-16  2025-07-29 13:45:51  2025-07-29 13:45:51         NaN    4350   \n",
       "1        2019  2025-07-29 13:45:51  2025-07-29 13:45:51         NaN  91-119   \n",
       "\n",
       "   Num Pages Issue Volume  Number Of Volumes Journal Abbreviation  \\\n",
       "0        NaN     1     12                NaN           Nat Commun   \n",
       "1        NaN   NaN     88                NaN        Adv Clin Chem   \n",
       "\n",
       "   Short Title  Series  Series Number  Series Text  Series Title  Publisher  \\\n",
       "0          NaN     NaN            NaN          NaN           NaN        NaN   \n",
       "1          NaN     NaN            NaN          NaN           NaN        NaN   \n",
       "\n",
       "   Place Language                                     Rights  Type  Archive  \\\n",
       "0    NaN      eng                     © 2021. The Author(s).   NaN      NaN   \n",
       "1    NaN      eng  © 2019 Elsevier Inc. All rights reserved.   NaN      NaN   \n",
       "\n",
       "  Archive Location  Library Catalog  Call Number  \\\n",
       "0              NaN              NaN          NaN   \n",
       "1              NaN              NaN          NaN   \n",
       "\n",
       "                                              Extra  Notes  File Attachments  \\\n",
       "0  Place: England PMID: 34272381  PMCID: PMC8285412    NaN               NaN   \n",
       "1               Place: United States PMID: 30612608    NaN               NaN   \n",
       "\n",
       "   Link Attachments  \\\n",
       "0               NaN   \n",
       "1               NaN   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                        Manual Tags  \\\n",
       "0                 *Genetic Predisposition to Disease; Biomarkers; Creatinine/blood; Cystatins/pharmacology; Databases, Genetic; Europe; Gene Expression Regulation/genetics; Genome-Wide Association Study; Glomerular Filtration Rate/*genetics; Humans; Kidney/*metabolism/physiology; Organ Specificity; Quantitative Trait Loci; Renal Insufficiency, Chronic/genetics/*metabolism; Risk Factors; RNA-Seq; Single-Cell Analysis   \n",
       "1  Acetylglucosaminidase/analysis; Animals; Arginine/analogs & derivatives/analysis; beta 2-Microglobulin/analysis; Biomarkers/analysis; Creatinine/analysis; Cystatin C; Cystatin C/analysis; GFR; Glomerular Filtration Rate; Hepatitis A Virus Cellular Receptor 1/analysis; Humans; Kidney/physiopathology; KIM-1; Lipocalin-2/analysis; miRNA; NGAL; Renal Insufficiency, Chronic/*diagnosis/physiopathology; β2-microglobulin   \n",
       "\n",
       "   Automatic Tags Editor  Series Editor  Translator  Contributor  \\\n",
       "0             NaN    NaN            NaN         NaN          NaN   \n",
       "1             NaN    NaN            NaN         NaN          NaN   \n",
       "\n",
       "   Attorney Agent  Book Author  Cast Member  Commenter  Composer  Cosponsor  \\\n",
       "0             NaN          NaN          NaN        NaN       NaN        NaN   \n",
       "1             NaN          NaN          NaN        NaN       NaN        NaN   \n",
       "\n",
       "   Counsel  Interviewer  Producer  Recipient  Reviewed Author  Scriptwriter  \\\n",
       "0      NaN          NaN       NaN        NaN              NaN           NaN   \n",
       "1      NaN          NaN       NaN        NaN              NaN           NaN   \n",
       "\n",
       "   Words By  Guest  Number  Edition  Running Time  Scale  Medium  \\\n",
       "0       NaN    NaN     NaN      NaN           NaN    NaN     NaN   \n",
       "1       NaN    NaN     NaN      NaN           NaN    NaN     NaN   \n",
       "\n",
       "   Artwork Size  Filing Date  Application Number  Assignee  Issuing Authority  \\\n",
       "0           NaN          NaN                 NaN       NaN                NaN   \n",
       "1           NaN          NaN                 NaN       NaN                NaN   \n",
       "\n",
       "   Country  Meeting Name Conference Name  Court  References  Reporter  \\\n",
       "0      NaN           NaN             NaN    NaN         NaN       NaN   \n",
       "1      NaN           NaN             NaN    NaN         NaN       NaN   \n",
       "\n",
       "   Legal Status  Priority Numbers  Programming Language  Version  System  \\\n",
       "0           NaN               NaN                   NaN      NaN     NaN   \n",
       "1           NaN               NaN                   NaN      NaN     NaN   \n",
       "\n",
       "   Code  Code Number  Section  Session  Committee  History  Legislative Body  \\\n",
       "0   NaN          NaN      NaN      NaN        NaN      NaN               NaN   \n",
       "1   NaN          NaN      NaN      NaN        NaN      NaN               NaN   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 combined_text  \\\n",
       "0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            discovery and prioritization of variants and genes for kidney function in >1.2 million individuals. genes underneath signals from genome-wide association studies (gwas) for kidney function are promising targets for functional studies, but prioritizing variants  and genes is challenging. by gwas meta-analysis for creatinine-based estimated  glomerular filtration rate (egfr) from the chronic kidney disease genetics  consortium and uk biobank (n = 1,201,909), we expand the number of egfrcrea loci  (424 loci, 201 novel; 9.8% egfrcrea variance explained by 634 independent signal  variants). our increased sample size in fine-mapping (n = 1,004,040, european)  more than doubles the number of signals with resolved fine-mapping (99% credible  sets down to 1 variant for 44 signals, ≤5 variants for 138 signals).  cystatin-based egfr and/or blood urea nitrogen association support 348 loci  (n = 460,826 and 852,678, respectively). our customizable tool for gene  prioritisation reveals 23 compelling genes including mechanistic insights and  enables navigation through genes and variants likely relevant for kidney function  in human to help select targets for experimental follow-up.   \n",
       "1  novel glomerular filtration markers. chronic kidney disease is currently assessed by estimated glomerular filtration rate, a mathematical construct based on creatinine or creatinine and cystatin  concentration. creatinine-based equations have improved with standardization  efforts and the modification of diet in renal disease study (mdrd) and  ckd-epidemiology collaboration study (ckd-epi). because the measurement of  creatinine is subject to interference from non-gfr determinants, alternative  markers have long been sought. these have included cystatin c and low molecular  weight proteins like β2-microglobulin and beta trace protein. tubular disease  often occurs before glomerular filtration is impaired and investigators have  investigated the excretion of other low molecular weight proteins such as  neutrophil gelatinase-associated lipocalin (ngal) and kidney injury molecule-1  and n-acetyl-β-d-glucosaminidase. while preliminary, there is some evidence  linking these analytes with gfr, disease stage and mortality. although  asymmetrical dimethyl arginine, an inhibitor of nitric oxide, has been shown to  be associated with progression of renal disease, symmetric dimethyl arginine may  be a better marker. recent work has also explored the potential of microrna  (mirna) analysis and metabolomics studies to further elucidate this complex  pathophysiologic disease process. investigators hope to improve our ability to  detect ckd by the use of test panels, i.e., various marker combinations thereof.  unfortunately, most of these markers lack standardization unlike traditional  measures that rely on creatinine and cystatin c measurement.   \n",
       "\n",
       "  decision  has_abstract  ADPKD?  OMICS?  BIOINFO?  HUMAN DATA? BIOMARKER?  \\\n",
       "0     None          True   False   False     False        False       None   \n",
       "1     None          True   False   False     False        False       None   \n",
       "\n",
       "  TRASLATIONAL? DECISION COMMENT  \n",
       "0          None     None    None  \n",
       "1          None     None    None  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# añadir columnas checkbox\n",
    "cols = [\"ADPKD?\", \"OMICS?\", \"BIOINFO?\", \"HUMAN DATA?\", \"BIOMARKER?\", \"TRASLATIONAL?\", \"DECISION\", \"COMMENT\"]\n",
    "for col in cols[:-4]:\n",
    "    df[col] = False\n",
    "# últimas 2 cols\n",
    "for col in cols[-4:]:\n",
    "    df[col] = None\n",
    "\n",
    "# check\n",
    "df.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cffef6a9-9b89-4b6f-bacc-64d31b40cc53",
   "metadata": {},
   "source": [
    "# 4.exportar todo\n",
    "y empezar screening en siguiente script"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "12b126fc-93eb-48a0-879a-751a5ff15155",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generado 'articulos_para_screening.csv' para screening\n",
      "Shape: (97, 98)\n",
      "Generado 'stats.csv' para screening\n",
      "Shape: (11, 4)\n"
     ]
    }
   ],
   "source": [
    "df.to_csv('Anexo_3_carpeta_1_inputs/tabla_3.3.1_articulos_para_screening.csv', index=False)\n",
    "print(\"Generado 'articulos_para_screening.csv' para screening\")\n",
    "print(f\"Shape: {df.shape}\")\n",
    "\n",
    "stats.to_csv('Anexo_3_carpeta_1_inputs/tabla_3.3.2.stats.csv', index=True)\n",
    "print(\"Generado 'stats.csv' para screening\")\n",
    "print(f\"Shape: {stats.shape}\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
